modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/dto/v1_0/CartItemDTOConverter.java

```diff
@@ -95,6 +98,24 @@ public CartItem toDTO(DTOConverterContext dtoConverterContext)
 		};
 	}
 
+	private String _getErrorMessage(
+		CommerceOrderItem commerceOrderItem, Locale locale) {
+
+		String error = null;
+
+		CPInstance cpInstance = commerceOrderItem.fetchCPInstance();
+
+		ResourceBundle resourceBundle = LanguageResources.getResourceBundle(
+			locale);
+
+		if (cpInstance == null) {
+			error = LanguageUtil.get(
+				resourceBundle, "the-product-is-no-longer-available");
+		}
```

Comment:
@luca-pellizzon we can move `resourceBundle inside the conditional since its only being used inside there.